tests/integration/utils/kube/kube_client.go

```diff
@@ -0,0 +1,171 @@
+package kube
+
+import (
+	"context"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v2"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/kiali/kiali/config"
+	"github.com/kiali/kiali/log"
+	"github.com/kiali/kiali/tools/cmd"
+)
+
+func NewDynamicClient(t *testing.T) dynamic.Interface {
+	t.Helper()
+
+	cfg, err := cmd.GetKubeConfig()
+	if err != nil {
+		t.Fatalf("Error getting kube config: %v", err)
+	}
+
+	client, err := dynamic.NewForConfig(cfg)
+	if err != nil {
+		t.Fatalf("Error creating dynamic client: %v", err)
+	}
+
+	return client
+}
+
+func NewKubeClient(t *testing.T) kubernetes.Interface {
+	t.Helper()
+
+	cfg, err := cmd.GetKubeConfig()
+	if err != nil {
+		t.Fatalf("Error getting kube config: %v", err)
+	}
+
+	client, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		t.Fatalf("Error creating kube client: %v", err)
+	}
+
+	return client
+}
+
+func DeleteKialiPod(ctx context.Context, kubeClient kubernetes.Interface, namespace string, currentKialiPod string) error {
+	log.Debugf("Deleting Kiali pod %s", currentKialiPod)
+	err := kubeClient.CoreV1().Pods(namespace).Delete(ctx, currentKialiPod, metav1.DeleteOptions{})
+	if err != nil {
+		log.Errorf("Error deleting Kiali pod %s", err)
+		return err
+	}
+	return nil
+}
+
+// Deletes the existing kiali Pod and waits for the new one to be ready.
+func RestartKialiPod(ctx context.Context, kubeClient kubernetes.Interface, namespace string, currentKialiPod string) error {
+	log.Debugf("Restarting kiali pod %s %s", namespace, currentKialiPod)
```

Comment:
nit: since this doesn't delete anymore can we call it something else and update or remove the comment/log statement?

Reply:
Good catch! Updated